runtime/lib/ttnn/program.cpp

```diff
@@ -67,17 +67,23 @@ static void
 run(::tt::target::ttnn::EltwiseOp const *op, ::ttnn::Device &device,
     std::unordered_map<std::uint32_t, ::ttnn::Tensor *> &liveTensors,
     std::list<::ttnn::Tensor> &tensorPool) {
+  assert(op->ins()->size() == 2 && "Unsupported number of inputs");
```

Comment:
I think we want the assert to live inside of the case. `::tt::target::ttnn::EltwiseOp` is intended to be generic for unaries, binaries, and narys.

Reply:
It'll be good to outline how we plan to group them in that case. Just to avoid confusion. E.g. do we want to match a similar structure as PyBuda has? Or we'll make some structural changes on top of it? 

<img width="569" alt="image" src="https://github.com/user-attachments/assets/1d4c7a09-e493-4ed0-8841-1d3a2656f703">

Either way, we should define the basic layout when we start with adding more ops. In this case, it'll be cleaner to add one more class layer to separate unaries, binaries, and narys that will inherit the current EltwiseOp class.

Reply:
Yeah, I agree with your points, for now, I will revert this assert check in switch cases, but let's think about what we should do for the next milestone.